### PR TITLE
SBT: shade geny and fastparse modules

### LINKED ISSUE #3106: Remove dependency to geny < 1.0.0 or shading geny
Related issues

- https://github.com/scalameta/metals/pull/5082
- https://github.com/scalameta/metals/pull/4770

Current scalameta parser depends on forked version of fastparse, which depends on geny 0.6.5.
Since, geny < 1.0.0, and geny 1.0.0 are binary incompatible, apps that depends on `scalameta` cannot update have `geny >= 1.0.0`.

> > Not sure how much scalameta fork is aprt from the original one, but it would be awesone if wr can go back to li-hayoi fastparse, I'll take a quick look
> We did that to maintain a binary compatibility and avoid dependency issues such as with Geny, so there are three solutions here:
> - shade geny also
> - remove geny from scalameta's fastparse
> - shade it automatically
https://github.com/scalameta/metals/pull/5082#issuecomment-1494265720

So, isn't going back to li-haoyi/fastparse 3.0.1 (that depends on geny 1.0.0) and bump scalameta/scalameta as 5.0.0 (because it's not binary compatible by geny) an option?

### PR DESCRIPTION
**Pull request by kitbellew**
Fixes #3106.

### CODE REVIEW

**Review by tgodzik — APPROVED** [review]
I am not sure why we never used shading, I assumed it was because it was flaky, but it seems to work here ok. I checked on the Metals workspace to be 100% sure, but I don't see any issues.

File: scalameta/contrib/shared/src/main/scala/scala/meta/contrib/DocToken.scala [hunk 1]
```diff
@@ -1,7 +1,7 @@
 package scala.meta.contrib
 
-import scala.meta.internal.fastparse._
```
[the remark this answers is not in the document]
Comment (kitbellew) [reply scalameta/contrib/shared/src/main/scala/scala/meta/contrib/DocToken.scala:3]: not sure i understand the question. in addition to shading in the first commit, I'm also replacing our copy (fastparse-v2) with the actual fastparse... was there any other reason to have that copy other than shading?

[the remark this answers is not in the document]
Comment (tgodzik) [reply scalameta/contrib/shared/src/main/scala/scala/meta/contrib/DocToken.scala:3]: nvm, that's an import not a package statement. I am oficially blind

**Comment by bjaglin** [comment]
For the record, I haven't observed it in Scalafix (and [I'll make sure we won't](https://github.com/scalacenter/scalafix/pull/1819#issuecomment-1636118496)), but I think this breaks binary compatibility: source built against older 4.8.x scalameta versions will fail if ran with newer patch versions.

**Comment by kitbellew** [comment]
@bjaglin could you explain, please?

**Comment by bjaglin** [comment]
Apologies, I assumed (too quickly) that these artifacts were used in some macro implementation (in which case code compiled against <4.8.3 would fail if running against >=4.8.3), but it may not be the case. So it would only be "breaking" for scalameta users that would explicitly, accidentally depend on these transitive dependencies - totally fine!

**Pull request merged**